test/unit/org/apache/cassandra/tools/nodetool/InvalidateCIDRPermissionsCacheTest.java

```diff
@@ -118,10 +115,13 @@ public void testInvalidateSingleCidrPermission()
         // ensure cidr permission is cached
         role.hasAccessFromIp(ipAddr);
         assertThat(originalReadsCount).isEqualTo(getCidrPermissionsReadCount());
+        assertThat(((CassandraCIDRAuthorizer) DatabaseDescriptor.getCIDRAuthorizer()).has(ROLE_A)).isTrue();
 
         // invalidate cidr permission
         ToolRunner.ToolResult tool = ToolRunner.invokeNodetool("invalidatecidrpermissionscache", ROLE_A.getRoleName());
         tool.assertOnCleanExit();
+        assertThat(tool.getStdout()).contains("Invalidated the role role_a from CIDR permissions cache");
```

Comment:
@ekaterinadimitrova2 @yifan-c As we set expiry on cache entries, it could be possible that while tests run on circleCI/Jenkins etc, there may be delay after the entry is loaded into the cache and before invalidate runs, resulting in entry being expired before the actual invalidation. In that case, checks at line 118 and 123 may fail, causing flaky test.

Reply:
You are right, and this repeated run proves it - very rarely, but it can be flaky - https://app.circleci.com/pipelines/github/ekaterinadimitrova2/cassandra/2496/workflows/6ec742b1-7b6d-4ea9-b81a-36486da34e14/jobs/39036

One way is probably to use the Util#flakyTest to rerun it once if it fails?
Another is now when we have https://github.com/apache/cassandra/pull/2648/files#diff-02dc3c7a2a64aef7633ac085c24deddfed1b32b469c9958f2c04d5e228e6144fR42 to rely on that message but that one is not always generated and we don't check the particular roles were invalidated. Open for other ideas, too.

Reply:
`I believe that should be sufficient in the unit test. `
My only concern is that it wasn't to catch the bug we found here

Reply:
Found a way. We can set high validity for cache, in the setup() before cache gets created. Rest of the setup() code remains as it is.

```
    @BeforeClass
    public static void setup() throws Exception
    {
        DatabaseDescriptor.setRolesValidity(Integer.MAX_VALUE-1);

```

Once we do that, cache's expireAfterWrite gets set to 2147483646 ms, which is 596+ hours, i.e, 24+ days. 

- We do not need has() function. We can just verify that `assertThat(tool.getStdout()).contains("Invalidated the role role_a from CIDR permissions cache")` after invalidating a role in the test testInvalidateSingleCidrPermission.

- Next unit test testInvalidateAllCidrPermissions doesn't go through the code path we are changing as we discussed in thread https://github.com/apache/cassandra/pull/2648/files#r1321766712

I will run the test under circle CI and let you know how it goes.

Reply:
Great, let me know how it goes. Can you also post a link to your patch so I can consolidate everything for the final patch, please? Thank you in advance.

Reply:
@ekaterinadimitrova2 here is the patch https://github.com/skoppu22/cassandra/pull/1/files